src/mount/conf.cc

```diff
@@ -67,7 +67,13 @@ extern "C" void mount_ceph_get_config_info(const char *config_file,
     }
 
     std::string addr;
+    if (eaddr.u.sa.sa_family == AF_INET6) {
+      addr += "[";
+    }
     addr += eaddr.ip_only_to_str();
+    if (eaddr.u.sa.sa_family == AF_INET6) {
+      addr += "]";
+    }
     addr += ":";
     addr += std::to_string(eaddr.get_port());
```

Comment:
I think the whole thing could be replaced with
```
std::ostringstream oss;
...
    if (oss.tellp() > 0) {
      oss << ",";
    }
    oss << eaddr.get_sockaddr();
...
monaddrs = oss.str();
```
This would handle brackets, colon, port, etc and be consistent with krbd.

Reply:
Good point! Applied this change in the latest commit.

Reply:
Please squash into a single commit named "mount: fix IPv6 parsing error when using the configuration file".